### PR TITLE
"Clean up" redirected responses

### LINKED ISSUE #220: Allow to configure redirect mode for fetch
For our site template project (for Google Developer Group events), we would like to allow add the settings redirect: 'follow' to the fetch method in the 'install' handler.

This should solve our issue with wrong redirects, as described for our site here:
https://github.com/gdg-x/hoverboard/issues/149
and on stackoverflow 
http://stackoverflow.com/a/40277730/243599

Not sure how the configuration could look like in the sw-precache file though. Maybe
```
fetchInit = {redirect : 'follow'}
```
It is only considered if handleFetch is true.

### PR DESCRIPTION
**Pull request by jeffposnick**
R: @addyosmani @gauntface 

This handles a new security restriction around redirected responses that is coming in Chrome 59, and is already implemented in Firefox. See https://bugs.chromium.org/p/chromium/issues/detail?id=669363&desc=2#c1 for more details.

Fixes #220

### CODE REVIEW

File: lib/functions.js [hunk 1]
```diff
@@ -77,5 +77,30 @@ module.exports = {
...
+
+    // Firefox 50 and below doesn't support the Response.body stream, so we may
+    // need to read the entire body to memory as a Blob.
+    var bodyPromise = 'body' in originalResponse ?
+      Promise.resolve(originalResponse.body) :
+      originalResponse.blob();
+
+    return bodyPromise.then(function(body) {
+      // new Response() is happy when passed either a stream or a Blob.
+      return new Response(body, {  // eslint-disable-line no-undef
```
Comment (gauntface) [inline lib/functions.js]: Would adding /* eslint-env browser */ fix the need for this eslint-disable?

File: test/test.js [hunk 2]
```diff
@@ -468,3 +468,43 @@ describe('generateRuntimeCaching', function() {
...
+  var globalResponse = global.Response;
+
+  before(function() {
+    if (!globalResponse) {
+      global.Response = require('node-fetch').Response;
+    }
+  });
+
+  it('should return the same response when redirected is false', function() {
+    var originalResponse = new Response(responseText); // eslint-disable-line no-undef
```
Comment (gauntface) [inline test/test.js]: could you use new global.Response() here?

File: lib/functions.js (hunk 1 above)
Comment (jeffposnick) [inline lib/functions.js]: I've added `worker`, since that's more accurate than `browser`. But yeah, it takes care of it.

File: test/test.js (hunk 2 above)
Comment (jeffposnick) [inline test/test.js]: Yup, switching.

**Pull request merged**